src/mono/netcore/System.Private.CoreLib/src/System/Runtime/Loader/AssemblyLoadContext.Mono.cs

```diff
@@ -22,6 +21,7 @@ internal IntPtr NativeALC
             }
         }
 
+        [DynamicDependency(nameof(_nativeAssemblyLoadContext))]
```

Comment:
Is this needed? the 3-argument ALC constructor keeps this alive
https://github.com/dotnet/runtime/blob/ceaa8ae8f79875e1987342f4ca9d05f4ada8ada0/src/libraries/System.Private.CoreLib/src/System/Runtime/Loader/AssemblyLoadContext.cs#L102

And the default ALC has a static constructor, which I assume keeps the whole thing alive

https://github.com/dotnet/runtime/blob/ceaa8ae8f79875e1987342f4ca9d05f4ada8ada0/src/libraries/System.Private.CoreLib/src/System/Runtime/Loader/AssemblyLoadContext.cs#L806-L813

Reply:
It's strictly not needed but I added it to make the dependency clearer but I can remove it.